### ISSUE TITLE
Can not install Dask package

### ISSUE THREAD

**Issue by dbalabka** [python-poetry/poetry#7768]
- **Poetry version**: `>=1.4.0`
- **Python version**: `3.8`
- **OS version and name**: `Ubuntu 20.04.5 LTS`

- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) stable Poetry version, installed using a recommended method.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have consulted the [FAQ](https://python-poetry.org/docs/faq/) and [blog](https://python-poetry.org/blog/) for any relevant entries or release notes.
- [x] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option) and have included the output below.

## Issue
```
poetry init
poetry add dask
```
Leading to the following error:
```
❯ poetry add dask
Using version ^2023.3.2 for dask

Updating dependencies
Resolving dependencies... (0.0s)

  AssertionError



  at ~/.local/pipx/venvs/poetry/lib/python3.8/site-packages/poetry/mixology/incompatibility.py:113 in __str__
      109│         )
      110│
      111│     def __str__(self) -> str:
      112│         if isinstance(self._cause, DependencyCause):
    → 113│             assert len(self._terms) == 2
      114│
      115│             depender = self._terms[0]
      116│             dependee = self._terms[1]
      117│             assert depender.is_positive()

```

**Comment by dimbleby** [comment]
you have not provided your pyproject.toml, and you have not searched for duplicates

please close

**Comment by dbalabka** [comment]
> you have not searched for duplicates

@dimbleby, I searched the duplicates but didn't find anything useful. Still, taking into account your confidence that I didn't search enough, I repeated the attempt.

I found the issue is related to my problem:
https://github.com/python-poetry/poetry/issues/7501

The problem is that I'm using the same name as the project "dask" like the required project:
```
[tool.poetry]
name = "dask"
version = "0.1.0"
description = ""
authors = ["Dmitry Balabka <dmitry.balabka@ecentria.com>"]
readme = "README.md"

[tool.poetry.dependencies]
python = ">=3.8"


[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
```

It would be great if poetry reported such an issue properly, instead of an assertion error. I belive that it will save others' time.
